### PR TITLE
fix nightly test failures

### PR DESCRIPTION
**Pull request by payes**
This PR implements the following:
1. fix MayastorNodeReady() func
2. wait for device to appear after nvme login
3. fix log messages
Signed-off-by: Payes Anand <payes.anand@mayadata.io>

### CODE REVIEW

File: src/common/k8stest/util_replica.go [hunk 1]
```diff
@@ -46,7 +47,16 @@ func ChecksumReplica(initiatorIP string, targetIP string, uri string) (string, e
...
+	for retryCount < maxRetries {
+		devicePath, err = agent.Exec(initiatorIP, "nvme list")
+		if devicePath != "" && err == nil {
+			break
+		}
+		time.Sleep(5 * time.Second)
+		retryCount++
+	}
 
 	if err != nil {
```
Comment (chriswldenyer) [inline src/common/k8stest/util_replica.go:61]: maybe also check that devicePath != "" here

**Review by chriswldenyer — APPROVED** [review]
LGTM. As you suggested just now, we should use 'nvme list -o json'

File: src/MQ-2219-rc-reconciliation/utils.go [hunk 2]
```diff
@@ -195,7 +195,7 @@ func deleteFaultyDisk(node string) {
 	Expect(err).ToNot(HaveOccurred())
 	table := "dmsetup delete " + e2e_config.GetConfig().PoolDevice
 	err = agent.CreateFaultyDevice(*addr, e2e_config.GetConfig().PoolDevice, table)
-	Expect(err).ToNot(HaveOccurred(), "Failed to create faulty disk on node %s: ", addr)
+	Expect(err).ToNot(HaveOccurred(), "Failed to delete faulty disk on node %s: ", addr)
```
[the remark this answers is not in the document]
Comment (payes) [reply src/MQ-2219-rc-reconciliation/utils.go:198]: Actually its a delete operation.

**Review by rohan2794 — APPROVED** [review]

**Comment by payes** [comment]
bors merge

**Pull request merged**